Build Failure clif_Mail_read Pre-RE < 20150513

* **rAthena Hash**: 6b7d6e684127b3d01b2539c6eb8eb033d57a26a5

* **Client Date**: 20080528, but anything less than 20150513

* **Server Mode**: Pre-Renewal

* **Description of Issue**: 
  * Result: map-server fails to compile
  * Expected Result: map-server should compile
  * How to Reproduce: Set PACKETVER to anything lower than 
  * Official Information: N/A
  
* **Modifications that may affect results**: Lowered packetver and enable pre-re, no other changes

Error from clif.cpp::clif_Mail_read

Github build-test on ubuntu-latest, GCC 7:
```
clif.cpp: In function ‘void clif_Mail_read(map_session_data*, int)’:
clif.cpp:16245:24: error: ‘data’ was not declared in this scope
   if( item->nameid && (data = itemdb_exists(item->nameid)) != NULL ) {
                        ^~~~
clif.cpp:16245:24: note: suggested alternative: ‘atan’
   if( item->nameid && (data = itemdb_exists(item->nameid)) != NULL ) {
                        ^~~~
                        atan
make[1]: *** [Makefile:99: obj/clif.o] Error 1
make: *** [Makefile:53: map] Error 2
make[1]: Leaving directory '/home/runner/work/***/***/src/map'
Error: Process completed with exit code 2.
```


MS Visual Studio 2019:
![devenv_VrPqSsorYz](https://user-images.githubusercontent.com/4101952/188266326-3012cedc-1a9f-4499-84f1-f3095725ad89.png)

--- a/src/map/clif.cpp
+++ b/src/map/clif.cpp
@@ -16242,7 +16242,9 @@ void clif_Mail_read( struct map_session_data *sd, int mail_id ){
 		WFIFOL(fd,72) = 0;
 		WFIFOL(fd,76) = msg->zeny;
 
-		if( item->nameid && (data = itemdb_exists(item->nameid)) != NULL ) {
+		std::shared_ptr<item_data> data = item_db.find(item->nameid);
+
+		if( item->nameid && data != nullptr ) {
 			WFIFOL(fd,80) = item->amount;
 			WFIFOW(fd,84) = client_nameid( item->nameid );
 			WFIFOW(fd,86) = itemtype( item->nameid );

